### ISSUE TITLE
How should cache / preprocess heatmap response data look like?

### ISSUE THREAD

**Issue by iori-yja** [Netflix/flamescope#58]
Every time people access the flamescope's heatmap, the server processes the perf output data which makes the user awaited. So I am considering to cache preprocessed or first-returned data.

Structure of the response is quite simple;
```
{
    "columns": [0, 1, .....],
    "rows": [980, 960....],
    "values": [ [2, 2, ...], [...], [...], ....],
    "maxvalue": 16,
}
```

The most quick hack to cache it is storing in sessionStorage in local browser. Better idea is to save the data as a metadata beside actual data (In that case, we need to decide naming convention for metadata). Other idea is to store data in DB.

I am going to implement a cache anyway, but I wanted to ask this before starting, which is,  _what kind of cache is easier to accept for you?_ because I want to keep it upstream.

**Comment by spiermar** [comment]
When you mention cache, is that from going back and forth between the heatmap and flame graph visualization, or changing profiles?

**Comment by spiermar** [comment]
Preprocess in the other hand is a different problem. I definitely wouldn't do it in the browser session storage, given the size limits, but might be Ok to have a task to preprocess all heatmaps and store them on disk.

**Comment by iori-yja** [comment]
I agree with you on both points of avoiding using databases and browser storages for each reason. I will move on to this in a few weeks. Is it OK to be default behavior to store preprocessed data next to the profile?
